### PR TITLE
Sc updates in noise

### PR DESCRIPTION
**Pull request by velzevur**
This is the proctol part of PT [SC: move `updates` in off-chain tx to noise](https://www.pivotaltracker.com/story/show/165593720)

This is a complementary PR for aeternity/aeternity#2399

I also adapted all examples to be using the JSON-RPC API

### CODE REVIEW

File: channels/OFF-CHAIN.md [hunk 1]
```diff
@@ -143,6 +145,28 @@ reserved for unrecoverable failures only.
...
 ### `ping`/`pong`
 
+## Sub-messages
+
+The following serializations can be consistently used below.
+
+### Updates list
+
+Progess off-chain is achieved by one participant proposing a set of updates to
+be applied on top of the last state as well with the next state that is
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
be applied on top of the last state as well as the next state that is
```

File: channels/OFF-CHAIN.md [hunk 2]
```diff
@@ -534,12 +562,12 @@ Message code: 8
 
 Once `funding_locked` messages have been exchanged, the channel enters the
 `open` state. Changes to the off-chain state can be effected by sending an
-`update` message. The payload of this message must be a singly-signed
-off-chain state, where the `updates` element contains a list of operations
-to be performed on the previous state, and the `state_hash` is the aggregated
-root hash of the resulting state trees. The receiver must verify that the
-state is a valid outcome, then return it, co-signed, in an `update_ack`
-message.
+`update` message. It consists of a singly-signed off-chain transaction and
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
`update` message. It consists of a single-signed off-chain transaction and
```

File: channels/OFF-CHAIN.md [hunk 3]
```diff
@@ -534,12 +562,12 @@ Message code: 8
...
+an updates list that contains a list of operations to be performed on the
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
a list of updates list containing the operations to be performed on the
```

File: channels/OFF-CHAIN.md [hunk 4]
```diff
@@ -534,12 +562,12 @@ Message code: 8
...
+previous state. The `state_hash` of the off-chain transaction is the
+aggregated root hash of the resulting state trees. The receiver must verify
+that the state is a valid outcome, then return it, co-signed, in an
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
that the state is the correct outcome, then return it, co-signed, in an
```

File: channels/OFF-CHAIN.md [hunk 5]
```diff
@@ -613,11 +643,10 @@ be the last mutually signed state. The receiver does not reply.
 Message code: 11
 
 In order to deposit more funds into the channel, one party can initiate
-a `deposit_created` request. The payload is a singly signed
-`channel_deposit_tx` object, which includes the state hash and round of the
-next off-chain state, after applying a deposit operation with the expected
-amount. The initiating side can only deposit coins from its own on-chain
-account (same public key) to its own off-chain account in the channel state.
+a `deposit_created` request. It consists of singly signed `channel_deposit_tx`
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
a `deposit_created` request. It consists of a single-signed `channel_deposit_tx`
```

File: channels/OFF-CHAIN.md [hunk 6]
```diff
@@ -719,11 +748,9 @@ be the last mutually signed state. The receiver does not reply.
 Message code: 15
 
 In order to withdraw coins from the channel, one party can initiate
-a `withdraw_created` request. The payload is a singly signed
-`channel_withdraw_tx` object, which includes the state hash and round of the
-next off-chain state, after applying a withdrawal operation with the expected
-amount. The initiating side can only withdraw coins from its own off-chain
-account in the channel state (same public key) to its own on-chain account.
+a `withdraw_created` request. It consists of a singly signed
```
Comment (hanssv) [inline channels/OFF-CHAIN.md]:
```suggestion
a `withdraw_created` request. It consists of a single-signed
```

**Comment by lucafavatella** [comment]
> This is a complementary PR for [aeternity/aeternity#2399](https://github.com/aeternity/aeternity/pull/2399)

https://github.com/aeternity/aeternity/pull/2396 ?

File: serializations.md [hunk 7]
```diff
@@ -761,6 +762,14 @@ The channel off-chain transaction is not included directly in the transaction tr
 ]
 ```
 
+version 2, from Fortuna release
```
Comment (lucafavatella) [inline serializations.md]: I see:
* [Definition in code](https://github.com/aeternity/aeternity/blob/3cad5f6848198c7931cbec214c6066d8b1117868/apps/aechannel/src/aesc_offchain_tx.erl#L198L201).
* [Relation to protocol version](https://github.com/aeternity/aeternity/blob/3cad5f6848198c7931cbec214c6066d8b1117868/apps/aechannel/src/aesc_offchain_tx.erl#L246).

Ok.

File: channels/OFF-CHAIN.md [hunk 8]
```diff
@@ -143,6 +145,28 @@ reserved for unrecoverable failures only.
...
+The following serializations can be consistently used below.
+
+### Updates list
+
+Progress off-chain is achieved by one participant proposing a set of updates to
+be applied on top of the last state as well as the next state that is
+produced.
+A single off-chain update's serialization has the following structure:
+
+  name                  size (bytes)
```
Comment (lucafavatella) [inline channels/OFF-CHAIN.md]: Strange markup?

**Pull request merged**